lib/logstash/outputs/elasticsearch.rb

```diff
@@ -50,114 +49,18 @@
 # Keep in mind that a connection with keepalive enabled will
 # not reevaluate its DNS value while the keepalive is in effect.
 class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
-  attr_reader :client
+  require "logstash/outputs/elasticsearch/http_client"
```

Comment:
I solve this annoying problem by having a 'namespace.rb' that all things require that describes the namespace of modules that will be used by anything. It's not a great solution, so the solution you have here is just fine :)